feat(documentation): Add link to figma

imagoiq opened the pull request.

imagoiq commented:
@rouvenpost Can you please check all the links and the missing list of components (in the description of this PR), please?

oliverschuerch reviewed (changes requested):
- [x] Add link to Card-Button docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=2452-66025&mode=design&t=jADNra0puVcBoQ9V-0
- [x] Add link to Breadcrumb docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=21648-20076&mode=design&t=HksCTWa2MMccgMl4-0
- [x] Add link to Footer docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=26665-17211&mode=design&t=HksCTWa2MMccgMl4-0
- [x] Add link to Header docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=19536-26512&mode=design&t=HksCTWa2MMccgMl4-0
- [ ] Add link to Intranet-Header docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=28233-57868&mode=design&t=HksCTWa2MMccgMl4-0
- [ ] Add link to Modal docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=20215-22938&mode=design&t=HksCTWa2MMccgMl4-0
- [ ] Add link to Notification Overlay docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=22748-36043&mode=design&t=HksCTWa2MMccgMl4-0
- [ ] Add link to Typeahead docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=22981-41449&mode=design&t=HksCTWa2MMccgMl4-0
- [ ] Add link to Carousel docs: ?
- [ ] Add link to Collapsible docs: ?
- [ ] Add link to Subnavigation docs: ?
- [ ] Add link to Timepicker docs: ?

The docs with unknown (?) Figma url can be ignored, I've just added them for documentation purposes.

Housten we have a problem:
![image](https://github.com/swisspost/design-system/assets/9716662/332ce55f-1a5f-482b-aca4-ddbff654a3f7)

My suggestion:
- [ ] Shorten the button text from "Figma Design" to "Figma" and the "See archived documentation" to "Archived docs".
- [ ] Take into consideration to implement small buttons (`btn-sm`, if easy to do so, add the class only on small devices).
- [ ] Let content wrap on a new line if there is not enouth space.

imagoiq commented on packages/documentation/src/stories/components/accordion/accordion.docs.mdx:
```diff
@@ -5,7 +5,11 @@ import SampleCustomTrigger from './accordion-custom-trigger.sample?raw';
+<div className="docs-title">
+  # Accordion
+
+  <link-design of={JSON.stringify(accordionStories)}></link-design>
```
I didn't find another way. Perhaps there is?

oliverschuerch replied:
I thought, as this is working for the storybook <Meta> block, it should be possible also for us.

imagoiq replied:
Yep, but web components can only get string in their property, that might be the issue.

imagoiq commented on packages/documentation/src/shared/link-design/link-design.component.ts:
I try to avoid adding target blank everywhere as it has some issue with accessibility. Maybe we can talk about this in the roundtable?

oliverschuerch replied:
What about this solution: https://dev.to/cerchie/3-ways-to-make-target-blank-accessible-3lan#comment-1kk1k

imagoiq commented:
> * [x]  Add link to Card-Button docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=2452-66025&mode=design&t=jADNra0puVcBoQ9V-0
> 
>     * [x]  Add link to Breadcrumb docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=21648-20076&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Footer docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=26665-17211&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Header docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=19536-26512&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Intranet-Header docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=28233-57868&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Modal docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=20215-22938&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Notification Overlay docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=22748-36043&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [x]  Add link to Typeahead docs: https://www.figma.com/file/xZ0IW0MJO0vnFicmrHiKaY/Components-Post?type=design&node-id=22981-41449&mode=design&t=HksCTWa2MMccgMl4-0
> 
>     * [ ]  Add link to Carousel docs: ?
> 
>     * [ ]  Add link to Collapsible docs: ?
> 
>     * [ ]  Add link to Subnavigation docs: ?
> 
>     * [ ]  Add link to Timepicker docs: ?
> 
> 
> The docs with unknown (?) Figma url can be ignored, I've just added them for documentation purposes.
> 
> Housten we have a problem: ![image](https://private-user-images.githubusercontent.com/9716662/319125705-332ce55f-1a5f-482b-aca4-ddbff654a3f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTQwOTgsIm5iZiI6MTcxMjE1Mzc5OCwicGF0aCI6Ii85NzE2NjYyLzMxOTEyNTcwNS0zMzJjZTU1Zi0xYTVmLTQ4MmItYWNhNC1kZGJmZjY1NGEzZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMTQxNjM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTZmNGNiNzE0ZmE3ZDIwMmJjMzgxNDc0OTdkNzExNWVhMmIzNTBlZTc1NjNjMmI3OTdjMDI0ZDNjMDZmYTk3MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9bevlMO70EJoBa-HDk9CutnKYQji5FPsX2gELPpGZjY)
> 
> My suggestion:
> 
>     * [ ]  Shorten the button text from "Figma Design" to "Figma" and the "See archived documentation" to "Archived docs".
> 
>     * [ ]  Take into consideration to implement small buttons (`btn-sm`, if easy to do so, add the class only on small devices).
> 
>     * [x]  Let content wrap on a new line if there is not enouth space.

Thanks for the research.
I fixed the responsiveness. About your suggestions, I think we don't need to shorten the links, just wrap them. I added the word "design" as it might not be clear for someone who doesn't know what Figma is.

rouvenpost commented:
It would be nice if the Figma Link opens in a new Tab instead of the same Tab...
